mice-and-cats-java/src/main/java/foop/world/World.java

```diff
@@ -65,46 +73,64 @@ public void afterEntityMoved(HashSet<Player> players) {
             return;
         }
 
-        var cat = entities.get(0);
+
+        var cats = entities.stream().filter(entity -> entity.getType() == CAT).toList();
         for (Entity e : entities) {
-            if (cat != e && !e.isDead() && cat.getPosition().equals(e.getPosition()) && cat.isUnderground() == e.isUnderground()) {
+            if (!cats.contains(e)
```

Comment:
Building a list and calling contains on it is a bit inefficient. It would be faster to just check that `e.getType() == MOUSE`.